remote-view/shared/src/main/scala/parsley/debug/RemoteView.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2023 Parsley Debug View Contributors <https://github.com/j-mie6/parsley-debug-views/graphs/contributors>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+package parsley.debug
+
+import parsley.debug.internal.DebugTreeSerialiser
+
+import sttp.client3.*
+
+import scala.util.Try
+import scala.concurrent.duration.*
+
+/** The RemoteView HTTP module allows the parsley debug tree to be passed off to a server through a specified port on
+  * local host (by default) or to a specified IP address. This enables all of the debug tree parsing, serving and 
+  * graphics to be done in a separate process, and served reactively to a client.
+  * 
+  * This module is part of the Debugging Interactively in parsLey Library (DILL) project,
+  * (https://github.com/j-mie6/parsley-debug-app).
+  * 
+  * RemoteView uses the STTP library to create HTTP requests to a specified IP address, over a specified port.
+  * The request is formatted using the upickle JSON formatting library, it is being used over other
+  * libraries like circe for its improved speed over large data structures.
+  */
+sealed trait RemoteView extends DebugView.Reusable {
+  implicit protected val port: Integer
```

Comment:
```suggestion
protected val port: Int
```
No need for these to be `implicit` (same with address). `Int` is used not `Integer`, which is a javaism

Reply:
👍

Reply:
👍